csrc/ir/nodes.cpp

```diff
@@ -2374,8 +2374,12 @@ std::vector<PolymorphicValue> ViewOp::evaluate(
   std::vector<int64_t> out_shape;
   out_shape.reserve(out_rfactor.size());
   for (IterDomain* id : out_rfactor) {
-    out_shape.push_back(
-        ee.evaluate(id->getMaybeExpandedExtent()).as<int64_t>());
+    if (id->isDeviceDim()) {
+      out_shape.push_back(1);
+    } else {
+      out_shape.push_back(
+          ee.evaluate(id->getMaybeExpandedExtent()).as<int64_t>());
+    }
```

Comment:
That is a good question. I don't quite know the answer, but my guess is the `ViewOp` is evaluated by FusionExecutor because the reshape is the output of a segment, since the sum is over a DID axis (maps to reduce scatter)
```
  std::vector<int64_t> orig_size = {K, M, N};
  std::vector<int64_t> new_size = {K, Mo, Mi, N};
  TensorView* c1 = reshape(c0, orig_size, new_size);
  TensorView* c = sum(c1, {0});
```